odoo/addons/base/ir/ir_mail_server.py

```diff
@@ -138,6 +138,11 @@ class IrMailServer(models.Model):
                           "specified for outgoing emails (To/Cc/Bcc)")
 
     name = fields.Char(string='Description', required=True, index=True)
+    smtp_from = fields.Char(
```

Comment:
Can you avoid to add a new field? Stable policy avoids to change DB layout in a stable version.

Reply:
@pedrobaeza about the only way I could think to do it is to have smtp_from a functional field with an inverse function that reads and writes an ir.config.parameter.  Say ir_mail_server.3.smtp_from for id 3. 

The other alternative and perhaps even better is to use a property field which won't add a column, but then obviously you have the multicompany management portion to sort out.

Reply:
Is adding a field against stable policy? I thought this applied to removing and changing fields only. 

I honestly don't like the other option, which would be to only have one outbound email address for the entire Odoo server vs. selecting per SMTP server.

Reply:
Yes, unfortunately you [can't add new stored fields](https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean) in stable versions, as it would crash production installations, should they simply pull the latest code and restart.
It's not pretty, but for stable versions we should probably use an `ir.config.parameter` entry to activate this workaround. And I would not event add a computed field at all, it's confusing (same value shared by all mail servers) and not strictly required. If we want to merge this in all stable versions (even 9.0), we should aim for the most minimal and less risky patch.
Anyhow I think a global toggle is enough, it should be quite sufficient in most cases.

Also, hardcoding the complete "smtp-from" does not seem to do what we want. In general the company owns a domain name, and any email that belongs to this domain should be let through untouched. This way we only alter the few emails that really do need to be altered.

The way I imagine it: when enabled via an ICP entry, this feature verifies that the "From" domain is a whitelisted one, otherwise it rewrites the "From" header using the "via" form, and replaces the actual email address with the bounce address returned by  `_get_default_bounce_address()`. This makes the feature consistent with our existing  mechanisms for handling replies. If we want a simpler alternative, instead of a whitelist we could only support the mail domain configured as "mail.catchall.domain".

We can get more fancy in master, but for stable the most important goal is to keep a minimal patch.